### PR TITLE
feat(core): Added flow for verifying unverified wallets

### PR DESCRIPTION
**Pull request by amanCypherock**

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4971 Wallet Verification from menu in case of Unverified valid wallet](https://app.clickup.com/t/9002019994/PRF-4971)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4971 Wallet Verification from menu in case of Unverified valid wallet](https://app.clickup.com/t/9002019994/PRF-4971)

File: src/wallet/verify_wallet_flow.c [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+  switch (current_state) {
+    case PIN_INPUT: {
+      if (!WALLET_IS_PIN_SET(wallet.wallet_info)) {
+        next_state = TAP_CARD_FLOW;
+        break;
+      }
+
+      WALLET_SET_PIN(wallet_for_flash.wallet_info);
+      WALLET_SET_PIN(wallet.wallet_info);
```
Comment (ujjwal-cyph) [inline src/wallet/verify_wallet_flow.c:144]: remove.

File: src/wallet/verify_wallet_flow.c [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+                 SHA256_DIGEST_LENGTH,
+                 wallet.password_double_hash);
+      memzero(flow_level.screen_input.input_text,
+              sizeof(flow_level.screen_input.input_text));
+      next_state = TAP_CARD_FLOW;
+      break;
+    }
+
+    case TAP_CARD_FLOW: {
+      if (true == card_flow_verify_wallet()) {
```
Comment (ujjwal-cyph) [inline src/wallet/verify_wallet_flow.c:167]: wrong PIN input can happen here. add the reasoning for that case; how we want to handle it.

**Review by ujjwal-cyph — CHANGES_REQUESTED** [review]
change according to comments

File: src/wallet/verify_wallet_flow.c (hunk 1 above)
Comment (amanCypherock) [reply src/wallet/verify_wallet_flow.c]: removed

File: src/menu/wallet_menu.c [hunk 3]
```diff
@@ -238,7 +240,18 @@ static void wallet_menu_handler(engine_ctx_t *ctx,
...
         case UNVERIFIED_VALID_WALLET:
-          // TODO: Handle verify wallet
+          verify_wallet_flow(wallet_ptr);
+
+          // If post verification, the wallet state was updated to INVALID,
+          // proceed to delete that wallet
+          if (INVALID_WALLET == wallet_ptr->state) {
+            message_scr_init(ui_text_wallet_verification_failed);
+            if (0 != get_state_on_confirm_scr(0, 1, 2)) {
+              return;
```
[the remark this answers is not in the document]
Comment (amanCypherock) [reply src/menu/wallet_menu.c]: updated

File: src/wallet/verify_wallet_flow.c (hunk 2 above)
Comment (amanCypherock) [reply src/wallet/verify_wallet_flow.c:164]: updated

Comment (amanCypherock) [reply src/wallet/verify_wallet_flow.c:164]: Only added error messages, not added retry attempt

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**